packages/react-tooltip/src/components/Tooltip/useTooltip.tsx

```diff
@@ -189,42 +189,38 @@ export const useTooltip_unstable = (props: TooltipProps): TooltipState => {
-    onBlur: useMergedEventCallbacks(child?.props?.onBlur, onLeaveTrigger),
-  };
-
-  // If the target prop is not provided, attach targetRef to the trigger element's ref prop
-  const childTargetRef = useMergedRefs(child?.ref, targetRef);
-  if (popperOptions.target === undefined) {
-    triggerProps.ref = childTargetRef;
-  }
+  const triggerAriaProps: Pick<TooltipTriggerProps, 'aria-label' | 'aria-labelledby' | 'aria-describedby'> = {};
 
   if (relationship === 'label') {
-    const hasLabel = child?.props && ('aria-label' in child.props || 'aria-labelledby' in child.props);
-    if (!hasLabel) {
-      // aria-label only works if the content is a string. Otherwise, need to use aria-labelledby.
-      if (typeof state.content.children === 'string') {
-        triggerProps['aria-label'] = state.content.children;
-      } else if (!isServerSideRender) {
-        triggerProps['aria-labelledby'] = state.content.id;
-        // Always render the tooltip even if hidden, so that aria-labelledby refers to a valid element
-        state.shouldRenderTooltip = true;
-      }
+    // aria-label only works if the content is a string. Otherwise, need to use aria-labelledby.
+    if (typeof state.content.children === 'string') {
+      triggerAriaProps['aria-label'] = state.content.children;
+    } else if (!isServerSideRender) {
+      triggerAriaProps['aria-labelledby'] = state.content.id;
+      // Always render the tooltip even if hidden, so that aria-labelledby refers to a valid element
+      state.shouldRenderTooltip = true;
     }
   } else if (relationship === 'description') {
-    const hasDescription = child?.props && ('aria-description' in child.props || 'aria-describedby' in child.props);
-    if (!hasDescription && !isServerSideRender) {
-      triggerProps['aria-describedby'] = state.content.id;
+    if (!isServerSideRender) {
+      triggerAriaProps['aria-describedby'] = state.content.id;
       // Always render the tooltip even if hidden, so that aria-describedby refers to a valid element
       state.shouldRenderTooltip = true;
     }
   }
 
+  const childTargetRef = useMergedRefs(child?.ref, targetRef);
+
   // Apply the trigger props to the child, either by calling the render function, or cloning with the new props
-  state.children = applyTriggerPropsToChildren(children, triggerProps);
+  state.children = applyTriggerPropsToChildren<TooltipTriggerProps>(children, {
+    ...triggerAriaProps,
+    ...child?.props,
+    // If the target prop is not provided, attach targetRef to the trigger element's ref prop
+    ref: popperOptions.target === undefined ? childTargetRef : child?.ref,
+    onPointerEnter: useMergedEventCallbacks(child?.props?.onPointerEnter, onEnterTrigger),
+    onPointerLeave: useMergedEventCallbacks(child?.props?.onPointerLeave, onLeaveTrigger),
+    onFocus: useMergedEventCallbacks(child?.props?.onFocus, onEnterTrigger),
+    onBlur: useMergedEventCallbacks(child?.props?.onBlur, onLeaveTrigger),
+  });
```

Comment:
Aligns order or merging and spreads `child.props` (https://github.com/microsoft/fluentui/pull/21495#discussion_r796392244) as we discussed in offline with @behowell & @ling1726.